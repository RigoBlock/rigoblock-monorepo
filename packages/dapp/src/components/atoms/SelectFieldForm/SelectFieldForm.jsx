import 'react-md/lib/Helpers'
import { Field } from 'redux-form'
import PropTypes from 'prop-types'
import React from 'react'
import SelectField from '../SelectField'

const SelectFieldForm = ({ fieldName, fieldProps }) => (
  <Field name={fieldName} component={SelectField} {...fieldProps} />
)

SelectFieldForm.propTypes = {
  fieldName: PropTypes.string.isRequired,
  fieldProps: PropTypes.object.isRequired
}

export default SelectFieldForm
