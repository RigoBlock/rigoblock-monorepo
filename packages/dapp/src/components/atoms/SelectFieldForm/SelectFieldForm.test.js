import { shallow } from 'enzyme'
import SelectFieldForm from './SelectFieldForm.jsx'
import toJson from 'enzyme-to-json'

const itemList = ['item1', 2, { label: 'test', value: '1' }]

const props = {
  fieldName: 'testField',
  fieldProps: {
    id: 1,
    items: itemList,
    onChange: jest.fn()
  }
}

describe('SelectFieldForm component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(SelectFieldForm, props)))
    ).toMatchSnapshot()
  })
})
