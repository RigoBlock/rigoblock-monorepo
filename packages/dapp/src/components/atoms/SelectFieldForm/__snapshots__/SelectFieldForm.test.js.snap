// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFieldForm component renders correctly 1`] = `
<Field
  component={[Function]}
  id={1}
  items={
    Array [
      "item1",
      2,
      Object {
        "label": "test",
        "value": "1",
      },
    ]
  }
  name="testField"
  onChange={[MockFunction]}
/>
`;
